### PR TITLE
added invite_user mailer functionality

### PR DESCRIPTION
**Pull request by alice-wang**
Added mail functionality for inviting users, added tests, fixed join method in user class, added unit test to reflect fix in user class

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -75,35 +75,39 @@ def delete_studygroup(studygroup_to_delete)
...
-        return_code = GlobalConstants::USER_ALREADY_IN_STUDYGROUP
+      unless Validation.user_in_studygroup(studygroup, user_to_invite)
+        code = GlobalConstants::USER_ALREADY_IN_STUDYGROUP
       end
 
-      # email users
-
-      return_code
+      mail = UserMailer.invite_email(self, user_to_invite, studygroup)
+      p '/'*10
```
Comment (dkumar) [inline app/models/user.rb]: Remove print

File: config/environments/development.rb [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+      :enable_starttls_auto => true,
+      :user_name => 'MeetMyClassMate@gmail.com',
+      :password => 'TeqHVEk1JyuvtTuDMsWNqYE8htrlloOmZ4s75xwytYh1G4Dj011OGkM09Igq7wr'
+  }
 
   #Devise purpose
-  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
-  config.action_mailer.delivery_method = :smtp
-  config.action_mailer.smtp_settings = {:address => 'localhost', :port => 1025}
+  # config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (dkumar) [inline config/environments/development.rb]: Remove unnecessary commented lines

**Comment by artsiriwatt** [comment]
looks dope body

File: spec/mailers/user_mailer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  # is the email being sent?
+  it 'should send an email' do
+    # ActionMailer::Base.deliveries.count.should == 1
+    expect(ActionMailer::Base.deliveries.count).to eq(1)
+  end
+
+  # is the email being sent to the desired recipient?
+  it 'renders the receiver email' do
+    # p '*'*10
```
Comment (dkumar) [inline spec/mailers/user_mailer_spec.rb]: Removed commented lines

File: spec/mailers/user_mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    UserMailer.invite_email(@owner, @user, @studygroup).deliver
+  end
+
+  after(:each) do
+    ActionMailer::Base.deliveries.clear
+  end
+
+  # is the email being sent?
+  it 'should send an email' do
+    # ActionMailer::Base.deliveries.count.should == 1
```
Comment (dkumar) [inline spec/mailers/user_mailer_spec.rb]: Remove commented lines

File: spec/mailers/user_mailer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  it 'renders the receiver email' do
+    # p '*'*10
+    # p ActionMailer::Base.deliveries.first.to.first
+    # ActionMailer::Base.deliveries.first.to.first.should == @user.email
+    expect(ActionMailer::Base.deliveries.first.to.first).to eq(@user.email)
+  end
+
+  # does the email have the correct subject line?
+  it 'should set the subject to the correct subject' do
+    # ActionMailer::Base.deliveries.first.subject.should == 'Invitation to join a studygroup'
```
Comment (dkumar) [inline spec/mailers/user_mailer_spec.rb]: Remove commented lines

**Comment by dkumar** [comment]
looks good

**Pull request merged**